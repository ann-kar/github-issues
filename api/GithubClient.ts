import fetch from "node-fetch";
import { GithubSearchResponse } from "../interfaces/github";

export class GithubClient {
  async getUsers(params: {
    perPage: number;
    page: number;
  }): Promise<GithubSearchResponse> {
    const { perPage, page } = params;
    const response = await fetch(
      "https://api.github.com/search/users?" +
        new URLSearchParams({
          q: "type:user",
          per_page: perPage.toString(),
          page: page.toString(),
        })
    );
    return (await response.json()) as GithubSearchResponse;
  }
}
