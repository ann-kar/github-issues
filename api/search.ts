import {
  ISearch,
  ISearchRequest,
  ISearchResponse,
  IUserSearchRequest,
  IUserSearchResponse,
} from "../interfaces/search";

export class Search implements ISearch {
  async search(request: ISearchRequest): Promise<ISearchResponse> {
    const response = await fetch(
      "http://localhost:3000/api/search?" +
        new URLSearchParams({
          query: request.query || "",
          page: request.page?.toString() || "",
          perPage: request.perPage?.toString() || "",
        })
    );
    return await response.json();
  }
  async userSearch(request: IUserSearchRequest): Promise<IUserSearchResponse> {
    throw new Error("Method not implemented.");
  }
}
