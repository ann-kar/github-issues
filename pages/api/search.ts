import { NextApiRequest, NextApiResponse } from "next";
import { GithubFacade } from "../../facades/GithubFacade";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  const data = await new GithubFacade().search({
    query: (req.query.query as string) || "",
    perPage: 10,
    page: 1,
  });
  res.status(200).json(data);
}
