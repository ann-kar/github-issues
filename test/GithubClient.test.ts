import { withRecording } from "./utils/withRecording";
import { GithubClient } from "../api/GithubClient";

describe("#getUsers", function () {
  test("returns github users with correct perPage", async () => {
    return withRecording(__dirname, "users", async () => {
      await expect(
        new GithubClient().getUsers({ perPage: 10, page: 1 })
      ).resolves.toMatchSnapshot();
    });
  });

  test("returns github users with correct page", async () => {
    return withRecording(__dirname, "users second page", async () => {
      await expect(
        new GithubClient().getUsers({ perPage: 10, page: 2 })
      ).resolves.toMatchSnapshot();
    });
  });
});
