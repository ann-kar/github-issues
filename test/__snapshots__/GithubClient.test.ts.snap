// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#getUsers returns github users with correct page 1`] = `
Object {
  "incomplete_results": true,
  "items": Array [
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/8547538?v=4",
      "events_url": "https://api.github.com/users/buckyroberts/events{/privacy}",
      "followers_url": "https://api.github.com/users/buckyroberts/followers",
      "following_url": "https://api.github.com/users/buckyroberts/following{/other_user}",
      "gists_url": "https://api.github.com/users/buckyroberts/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/buckyroberts",
      "id": 8547538,
      "login": "buckyroberts",
      "node_id": "MDQ6VXNlcjg1NDc1Mzg=",
      "organizations_url": "https://api.github.com/users/buckyroberts/orgs",
      "received_events_url": "https://api.github.com/users/buckyroberts/received_events",
      "repos_url": "https://api.github.com/users/buckyroberts/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/buckyroberts/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/buckyroberts/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/buckyroberts",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/80?v=4",
      "events_url": "https://api.github.com/users/ry/events{/privacy}",
      "followers_url": "https://api.github.com/users/ry/followers",
      "following_url": "https://api.github.com/users/ry/following{/other_user}",
      "gists_url": "https://api.github.com/users/ry/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/ry",
      "id": 80,
      "login": "ry",
      "node_id": "MDQ6VXNlcjgw",
      "organizations_url": "https://api.github.com/users/ry/orgs",
      "received_events_url": "https://api.github.com/users/ry/received_events",
      "repos_url": "https://api.github.com/users/ry/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/ry/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/ry/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/ry",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/5003903?v=4",
      "events_url": "https://api.github.com/users/StephenGrider/events{/privacy}",
      "followers_url": "https://api.github.com/users/StephenGrider/followers",
      "following_url": "https://api.github.com/users/StephenGrider/following{/other_user}",
      "gists_url": "https://api.github.com/users/StephenGrider/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/StephenGrider",
      "id": 5003903,
      "login": "StephenGrider",
      "node_id": "MDQ6VXNlcjUwMDM5MDM=",
      "organizations_url": "https://api.github.com/users/StephenGrider/orgs",
      "received_events_url": "https://api.github.com/users/StephenGrider/received_events",
      "repos_url": "https://api.github.com/users/StephenGrider/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/StephenGrider/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/StephenGrider/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/StephenGrider",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/7872329?v=4",
      "events_url": "https://api.github.com/users/benawad/events{/privacy}",
      "followers_url": "https://api.github.com/users/benawad/followers",
      "following_url": "https://api.github.com/users/benawad/following{/other_user}",
      "gists_url": "https://api.github.com/users/benawad/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/benawad",
      "id": 7872329,
      "login": "benawad",
      "node_id": "MDQ6VXNlcjc4NzIzMjk=",
      "organizations_url": "https://api.github.com/users/benawad/orgs",
      "received_events_url": "https://api.github.com/users/benawad/received_events",
      "repos_url": "https://api.github.com/users/benawad/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/benawad/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/benawad/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/benawad",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/463230?v=4",
      "events_url": "https://api.github.com/users/taylorotwell/events{/privacy}",
      "followers_url": "https://api.github.com/users/taylorotwell/followers",
      "following_url": "https://api.github.com/users/taylorotwell/following{/other_user}",
      "gists_url": "https://api.github.com/users/taylorotwell/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/taylorotwell",
      "id": 463230,
      "login": "taylorotwell",
      "node_id": "MDQ6VXNlcjQ2MzIzMA==",
      "organizations_url": "https://api.github.com/users/taylorotwell/orgs",
      "received_events_url": "https://api.github.com/users/taylorotwell/received_events",
      "repos_url": "https://api.github.com/users/taylorotwell/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/taylorotwell/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/taylorotwell/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/taylorotwell",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/54322854?v=4",
      "events_url": "https://api.github.com/users/rafaballerini/events{/privacy}",
      "followers_url": "https://api.github.com/users/rafaballerini/followers",
      "following_url": "https://api.github.com/users/rafaballerini/following{/other_user}",
      "gists_url": "https://api.github.com/users/rafaballerini/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/rafaballerini",
      "id": 54322854,
      "login": "rafaballerini",
      "node_id": "MDQ6VXNlcjU0MzIyODU0",
      "organizations_url": "https://api.github.com/users/rafaballerini/orgs",
      "received_events_url": "https://api.github.com/users/rafaballerini/received_events",
      "repos_url": "https://api.github.com/users/rafaballerini/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/rafaballerini/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/rafaballerini/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/rafaballerini",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/4248081?v=4",
      "events_url": "https://api.github.com/users/filipedeschamps/events{/privacy}",
      "followers_url": "https://api.github.com/users/filipedeschamps/followers",
      "following_url": "https://api.github.com/users/filipedeschamps/following{/other_user}",
      "gists_url": "https://api.github.com/users/filipedeschamps/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/filipedeschamps",
      "id": 4248081,
      "login": "filipedeschamps",
      "node_id": "MDQ6VXNlcjQyNDgwODE=",
      "organizations_url": "https://api.github.com/users/filipedeschamps/orgs",
      "received_events_url": "https://api.github.com/users/filipedeschamps/received_events",
      "repos_url": "https://api.github.com/users/filipedeschamps/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/filipedeschamps/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/filipedeschamps/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/filipedeschamps",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/1?v=4",
      "events_url": "https://api.github.com/users/mojombo/events{/privacy}",
      "followers_url": "https://api.github.com/users/mojombo/followers",
      "following_url": "https://api.github.com/users/mojombo/following{/other_user}",
      "gists_url": "https://api.github.com/users/mojombo/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/mojombo",
      "id": 1,
      "login": "mojombo",
      "node_id": "MDQ6VXNlcjE=",
      "organizations_url": "https://api.github.com/users/mojombo/orgs",
      "received_events_url": "https://api.github.com/users/mojombo/received_events",
      "repos_url": "https://api.github.com/users/mojombo/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/mojombo/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/mojombo/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/mojombo",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/4196?v=4",
      "events_url": "https://api.github.com/users/hadley/events{/privacy}",
      "followers_url": "https://api.github.com/users/hadley/followers",
      "following_url": "https://api.github.com/users/hadley/following{/other_user}",
      "gists_url": "https://api.github.com/users/hadley/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/hadley",
      "id": 4196,
      "login": "hadley",
      "node_id": "MDQ6VXNlcjQxOTY=",
      "organizations_url": "https://api.github.com/users/hadley/orgs",
      "received_events_url": "https://api.github.com/users/hadley/received_events",
      "repos_url": "https://api.github.com/users/hadley/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/hadley/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/hadley/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/hadley",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/2254731?v=4",
      "events_url": "https://api.github.com/users/diego3g/events{/privacy}",
      "followers_url": "https://api.github.com/users/diego3g/followers",
      "following_url": "https://api.github.com/users/diego3g/following{/other_user}",
      "gists_url": "https://api.github.com/users/diego3g/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/diego3g",
      "id": 2254731,
      "login": "diego3g",
      "node_id": "MDQ6VXNlcjIyNTQ3MzE=",
      "organizations_url": "https://api.github.com/users/diego3g/orgs",
      "received_events_url": "https://api.github.com/users/diego3g/received_events",
      "repos_url": "https://api.github.com/users/diego3g/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/diego3g/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/diego3g/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/diego3g",
    },
  ],
  "total_count": 80735811,
}
`;

exports[`#getUsers returns github users with correct perPage 1`] = `
Object {
  "incomplete_results": true,
  "items": Array [
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/1024025?v=4",
      "events_url": "https://api.github.com/users/torvalds/events{/privacy}",
      "followers_url": "https://api.github.com/users/torvalds/followers",
      "following_url": "https://api.github.com/users/torvalds/following{/other_user}",
      "gists_url": "https://api.github.com/users/torvalds/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/torvalds",
      "id": 1024025,
      "login": "torvalds",
      "node_id": "MDQ6VXNlcjEwMjQwMjU=",
      "organizations_url": "https://api.github.com/users/torvalds/orgs",
      "received_events_url": "https://api.github.com/users/torvalds/received_events",
      "repos_url": "https://api.github.com/users/torvalds/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/torvalds/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/torvalds/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/torvalds",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/905434?v=4",
      "events_url": "https://api.github.com/users/ruanyf/events{/privacy}",
      "followers_url": "https://api.github.com/users/ruanyf/followers",
      "following_url": "https://api.github.com/users/ruanyf/following{/other_user}",
      "gists_url": "https://api.github.com/users/ruanyf/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/ruanyf",
      "id": 905434,
      "login": "ruanyf",
      "node_id": "MDQ6VXNlcjkwNTQzNA==",
      "organizations_url": "https://api.github.com/users/ruanyf/orgs",
      "received_events_url": "https://api.github.com/users/ruanyf/received_events",
      "repos_url": "https://api.github.com/users/ruanyf/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/ruanyf/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/ruanyf/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/ruanyf",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/66577?v=4",
      "events_url": "https://api.github.com/users/JakeWharton/events{/privacy}",
      "followers_url": "https://api.github.com/users/JakeWharton/followers",
      "following_url": "https://api.github.com/users/JakeWharton/following{/other_user}",
      "gists_url": "https://api.github.com/users/JakeWharton/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/JakeWharton",
      "id": 66577,
      "login": "JakeWharton",
      "node_id": "MDQ6VXNlcjY2NTc3",
      "organizations_url": "https://api.github.com/users/JakeWharton/orgs",
      "received_events_url": "https://api.github.com/users/JakeWharton/received_events",
      "repos_url": "https://api.github.com/users/JakeWharton/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/JakeWharton/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/JakeWharton/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/JakeWharton",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/170270?v=4",
      "events_url": "https://api.github.com/users/sindresorhus/events{/privacy}",
      "followers_url": "https://api.github.com/users/sindresorhus/followers",
      "following_url": "https://api.github.com/users/sindresorhus/following{/other_user}",
      "gists_url": "https://api.github.com/users/sindresorhus/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/sindresorhus",
      "id": 170270,
      "login": "sindresorhus",
      "node_id": "MDQ6VXNlcjE3MDI3MA==",
      "organizations_url": "https://api.github.com/users/sindresorhus/orgs",
      "received_events_url": "https://api.github.com/users/sindresorhus/received_events",
      "repos_url": "https://api.github.com/users/sindresorhus/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/sindresorhus/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/sindresorhus/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/sindresorhus",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/25254?v=4",
      "events_url": "https://api.github.com/users/tj/events{/privacy}",
      "followers_url": "https://api.github.com/users/tj/followers",
      "following_url": "https://api.github.com/users/tj/following{/other_user}",
      "gists_url": "https://api.github.com/users/tj/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/tj",
      "id": 25254,
      "login": "tj",
      "node_id": "MDQ6VXNlcjI1MjU0",
      "organizations_url": "https://api.github.com/users/tj/orgs",
      "received_events_url": "https://api.github.com/users/tj/received_events",
      "repos_url": "https://api.github.com/users/tj/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/tj/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/tj/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/tj",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/8683378?v=4",
      "events_url": "https://api.github.com/users/gustavoguanabara/events{/privacy}",
      "followers_url": "https://api.github.com/users/gustavoguanabara/followers",
      "following_url": "https://api.github.com/users/gustavoguanabara/following{/other_user}",
      "gists_url": "https://api.github.com/users/gustavoguanabara/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/gustavoguanabara",
      "id": 8683378,
      "login": "gustavoguanabara",
      "node_id": "MDQ6VXNlcjg2ODMzNzg=",
      "organizations_url": "https://api.github.com/users/gustavoguanabara/orgs",
      "received_events_url": "https://api.github.com/users/gustavoguanabara/received_events",
      "repos_url": "https://api.github.com/users/gustavoguanabara/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/gustavoguanabara/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gustavoguanabara/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/gustavoguanabara",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/110953?v=4",
      "events_url": "https://api.github.com/users/addyosmani/events{/privacy}",
      "followers_url": "https://api.github.com/users/addyosmani/followers",
      "following_url": "https://api.github.com/users/addyosmani/following{/other_user}",
      "gists_url": "https://api.github.com/users/addyosmani/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/addyosmani",
      "id": 110953,
      "login": "addyosmani",
      "node_id": "MDQ6VXNlcjExMDk1Mw==",
      "organizations_url": "https://api.github.com/users/addyosmani/orgs",
      "received_events_url": "https://api.github.com/users/addyosmani/received_events",
      "repos_url": "https://api.github.com/users/addyosmani/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/addyosmani/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/addyosmani/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/addyosmani",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/1279609?v=4",
      "events_url": "https://api.github.com/users/llSourcell/events{/privacy}",
      "followers_url": "https://api.github.com/users/llSourcell/followers",
      "following_url": "https://api.github.com/users/llSourcell/following{/other_user}",
      "gists_url": "https://api.github.com/users/llSourcell/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/llSourcell",
      "id": 1279609,
      "login": "llSourcell",
      "node_id": "MDQ6VXNlcjEyNzk2MDk=",
      "organizations_url": "https://api.github.com/users/llSourcell/orgs",
      "received_events_url": "https://api.github.com/users/llSourcell/received_events",
      "repos_url": "https://api.github.com/users/llSourcell/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/llSourcell/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/llSourcell/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/llSourcell",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/150330?v=4",
      "events_url": "https://api.github.com/users/getify/events{/privacy}",
      "followers_url": "https://api.github.com/users/getify/followers",
      "following_url": "https://api.github.com/users/getify/following{/other_user}",
      "gists_url": "https://api.github.com/users/getify/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/getify",
      "id": 150330,
      "login": "getify",
      "node_id": "MDQ6VXNlcjE1MDMzMA==",
      "organizations_url": "https://api.github.com/users/getify/orgs",
      "received_events_url": "https://api.github.com/users/getify/received_events",
      "repos_url": "https://api.github.com/users/getify/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/getify/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/getify/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/getify",
    },
    Object {
      "avatar_url": "https://avatars.githubusercontent.com/u/119893?v=4",
      "events_url": "https://api.github.com/users/kennethreitz/events{/privacy}",
      "followers_url": "https://api.github.com/users/kennethreitz/followers",
      "following_url": "https://api.github.com/users/kennethreitz/following{/other_user}",
      "gists_url": "https://api.github.com/users/kennethreitz/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/kennethreitz",
      "id": 119893,
      "login": "kennethreitz",
      "node_id": "MDQ6VXNlcjExOTg5Mw==",
      "organizations_url": "https://api.github.com/users/kennethreitz/orgs",
      "received_events_url": "https://api.github.com/users/kennethreitz/received_events",
      "repos_url": "https://api.github.com/users/kennethreitz/repos",
      "score": 1,
      "site_admin": false,
      "starred_url": "https://api.github.com/users/kennethreitz/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/kennethreitz/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/kennethreitz",
    },
  ],
  "total_count": 76982390,
}
`;
